### PR TITLE
Priority feature

### PR DESCRIPTION
**Pull request by Joven-Heng**
Added Priority feature field, 

Currently working on UI

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -101,7 +109,7 @@ public boolean isSamePerson(Person otherPerson) {
      * If either objects are null, false is returned, as the other identity field
      * would be used to check for "sameness" instead.
      *
-     * @param obj First object to test for is same.
+     * @param obj      First object to test for is same.
```
Comment (WeiJie96) [inline src/main/java/seedu/address/model/person/Person.java:112]: Perhaps we should remove the extra tab here

File: src/main/java/seedu/address/model/person/Priority.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+/**
+ * Represents a Person's priority number in the client list.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPriority(String)}
+ */
+public class Priority {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Priority numbers should only contain characters L,M,H,U OR ulmh";
```
Comment (WeiJie96) [inline src/main/java/seedu/address/model/person/Priority.java:13]: Perhaps we should change to
"Priority values should only take L, M, H, U, l, m, h, u, low, medium, high, or undefined;

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -174,6 +183,11 @@ public String toString() {
                     .append(getNote());
         }
 
+        if (getPriority() != null) {
+            builder.append(" Priority: ")
```
Comment (WeiJie96) [inline src/main/java/seedu/address/model/person/Person.java:187]: Perhaps we dont need to check null here

File: src/main/java/seedu/address/model/person/Person.java [hunk 4]
```diff
@@ -131,7 +139,8 @@ public boolean equals(Object other) {
                 && equalsNullable(otherPerson.getEmail(), getEmail())
                 && equalsNullable(otherPerson.getAddress(), getAddress())
                 && otherPerson.getClientSources().equals(getClientSources())
-                && equalsNullable(otherPerson.getNote(), getNote());
+                && equalsNullable(otherPerson.getNote(), getNote())
+                && equalsNullable(otherPerson.getPriority(), getPriority());
```
Comment (WeiJie96) [inline src/main/java/seedu/address/model/person/Person.java:143]: Perhaps we dont need to use equalsNullable here

File: src/main/java/seedu/address/model/person/Person.java [hunk 5]
```diff
@@ -25,19 +25,23 @@
...
+    private final Priority priority;
 
 
     /**
      * Only name and tags need to be non-null.
      */
-    public Person(Name name, Phone phone, Email email, Address address, Set<ClientSource> clientSources, Note note) {
+    public Person(Name name, Phone phone, Email email, Address address, Set<ClientSource> clientSources, Note note,
+                  Priority priority) {
         requireAllNonNull(name, clientSources);
```
Comment (WeiJie96) [inline src/main/java/seedu/address/model/person/Person.java:36]: Perhaps we should require non null for priority

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 6]
```diff
@@ -148,7 +158,18 @@ public Person toModelType() throws IllegalValueException {
             modelNote = new Note(note);
         }
 
-        return new Person(modelName, modelPhone, modelEmail, modelAddress, modelClientSources, modelNote);
+        final Priority modelPriority;
+        if (priority == null) {
+            modelPriority = new Priority(null);
```
Comment (WeiJie96) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:163]: Perhaps we should throw error if priority is null

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 7]
```diff
@@ -85,6 +89,12 @@ public JsonAdaptedPerson(Person source) {
         } else {
             note = null;
         }
+
+        if (source.getPriority() != null) {
+            priority = source.getPriority().value;
+        } else {
+            priority = null;
+        }
```
Comment (WeiJie96) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:97]: Perhaps we dont need to check for null

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 8]
```diff
@@ -68,6 +70,10 @@ public PersonCard(Person person, int displayedIndex) {
             note.getChildren().add(new Label(person.getNote().noteName));
         }
 
+        if (person.getPriority() != null) {
```
Comment (WeiJie96) [inline src/main/java/seedu/address/ui/PersonCard.java:73]: Perhaps we dont need to check for null

File: src/test/java/seedu/address/testutil/PersonBuilder.java [hunk 9]
```diff
@@ -66,7 +71,7 @@ public PersonBuilder withName(String name) {
      * Parses the {@code clientSources} into a {@code Set<ClientSource>}
      * and set it to the {@code Person} that we are building.
      */
-    public PersonBuilder withClientSources(String ... clientSources) {
+    public PersonBuilder withClientSources(String... clientSources) {
```
Comment (WeiJie96) [inline src/test/java/seedu/address/testutil/PersonBuilder.java:74]: Perhaps we should add back the space here

File: src/main/java/seedu/address/model/person/Priority.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    public static final String VALIDATION_REGEX = "[ulmhULMH]|^undefined$|^high$|^medium$|^low$";
```
Comment (WeiJie96) [inline src/main/java/seedu/address/model/person/Priority.java:14]: The ^$ might not be necessary, but I might be wrong
"[ulmhULMH]|undefined|high|medium|low" might work

File: src/main/java/seedu/address/model/person/Priority.java [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        case "low":
+            return PriorityType.LOW.name().toLowerCase();
+        case "m":
+        case "medium":
+            return PriorityType.MEDIUM.name().toLowerCase();
+        case "h":
+        case "high":
+            return PriorityType.HIGH.name().toLowerCase();
+        default:
+            return "x"; // priority type does not exist
```
Comment (WeiJie96) [inline src/main/java/seedu/address/model/person/Priority.java:61]: Perhaps we should add an assertion here

**Review by WeiJie96 — CHANGES_REQUESTED** [review]
Overall good, just need to check for non-null for Priority in places other than AddCommand. I might have missed those out.

File: src/test/java/seedu/address/testutil/PersonBuilder.java (hunk 9 above)
Comment (Joven-Heng) [reply src/test/java/seedu/address/testutil/PersonBuilder.java:74]: uhm this string matters? But ok changed.

File: src/main/java/seedu/address/model/person/Priority.java (hunk 10 above)
Comment (Joven-Heng) [reply src/main/java/seedu/address/model/person/Priority.java:14]: Hmm but it works now why change it ><

File: src/main/java/seedu/address/model/person/Priority.java (hunk 11 above)
Comment (Joven-Heng) [reply src/main/java/seedu/address/model/person/Priority.java:61]: Ah my bad, initially this had some use as it was a parser but after changing it, it now doesnt.

**Review by WeiJie96 — APPROVED** [review]
LGTM

**Pull request merged**